lib/RowChecksum.pm

```diff
@@ -102,18 +102,30 @@ sub make_row_checksum {
       $sep    =~ s/'//g;
       $sep  ||= '#';
 
+      my @converted_cols;
+      for my $col(@{$cols->{select}}) {
+          my $colname = $col;
+          $colname =~ s/`//g;
+          my $type = $tbl_struct->{type_for}->{$colname} || '';
+          if ($type =~ m/^(CHAR|VARCHAR|BINARY|VARBINARY|BLOB|TEXT|ENUM|SET|JSON)$/i) {
```

Comment:
Should we include `ENUM`s?
Enums are handled internally as an index number so most probably this is correct, but I am wondering if a conversion might affect how the column is being handled.
This would be a very particular case but, we could add a specific test.

Reply:
It's already here: if ($type =~ m/^(CHAR|VARCHAR|BINARY|VARBINARY|BLOB|TEXT|**ENUM**|SET|JSON)$/i) {